docs/design/datacontracts/ECall.md

```diff
@@ -0,0 +1,39 @@
+# Contract ECall
+
+This contract is for fetching information related to native calls into the runtime.
+
+## APIs of contract
+
+``` csharp
+// Given an FCall entrypoint returns the corresponding MethodDesc.
+// If the address does not correspond to an FCall, returns TargetPointer.Null.
+TargetPointer MapTargetBackToMethodDesc(TargetCodePointer address);
+```
+
+## Version 1
+
+Global variables used
+| Global Name | Type | Purpose |
+| --- | --- | --- |
+| FCallMethods | ECHash[] | Hash table containing ECHash structures |
+| FCallHashSize | uint | Number of buckets in the hash table |
+
+
+Data descriptors used:
+| Data Descriptor Name | Field | Meaning |
+| --- | --- | --- |
+| `ECHash` | `Next` | Pointer to the next ECHash in the chain |
```

Comment:
ECHash is a left over from FCalls with helper method frames and other things that we do not have anymore. I am deleting it in https://github.com/dotnet/runtime/pull/117861